src/wasm-lib/kcl/src/execution/mod.rs

```diff
@@ -2020,6 +2073,22 @@ impl ExecutorContext {
         exec_state: &mut ExecState,
         body_type: BodyType,
     ) -> Result<Option<KclValue>, KclError> {
+        if let Some((annotation, source_range)) = program
+            .non_code_meta
+            .start_nodes
```

Comment:
If I'm reading this right, every Program can have `@settings()`. E.g. function bodies, if-else bodies. Is this intentional, or should we make an issue for future work?

Reply:
That's correct and it's kind-of intentional (in that I thought about it but then forgot to finish it off, so thanks for catching it!). It doesn't actually work correctly though because the old value isn't restored except for full modules. So we should have an issue to either make it work properly or to prohibit it.